.github/workflows/release.yaml

```diff
@@ -0,0 +1,188 @@
+        with:
+          name: falcon-${{ matrix.goos }}-${{ matrix.goarch }}
+          path: falcon-${{ matrix.goos }}-${{ matrix.goarch }}.tar.gz
+          if-no-files-found: error
+
+  release:
+    name: Create Release
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          fetch-depth: 0  # Fetch all history for changelog
+
+      - name: Download all artifacts
+        uses: actions/download-artifact@cc203385981b70ca67e1cc392babf9cc229d5806 # v4.1.9
+        with:
+          path: artifacts
+
+      - name: Prepare release assets
+        run: |
+          mkdir -p release
+          # Move all compressed binaries to release directory
+          find artifacts -type f -name 'falcon-*.tar.gz' -exec mv {} release/ \;
+          ls -lh release/
+
+      - name: Generate checksums
+        run: |
+          cd release
+          sha256sum falcon-*.tar.gz > checksums.txt
+          cat checksums.txt
+
+      - name: Generate changelog
+        id: changelog
+        run: |
+          # Get the previous tag
+          CURRENT_TAG=${GITHUB_REF#refs/tags/}
+          PREVIOUS_TAG=$(git describe --tags --abbrev=0 ${CURRENT_TAG}^ 2>/dev/null || echo "")
+
+          echo "Current tag: ${CURRENT_TAG}"
+          echo "Previous tag: ${PREVIOUS_TAG}"
+
+          # Generate changelog
+          if [ -z "${PREVIOUS_TAG}" ]; then
+            # First release - list all commits
+            echo "## Changes" > CHANGELOG.md
+            echo "" >> CHANGELOG.md
+            git log --pretty=format:"* %s (%h)" >> CHANGELOG.md
+          else
+            # List commits between previous and current tag
+            echo "## Changes since ${PREVIOUS_TAG}" > CHANGELOG.md
+            echo "" >> CHANGELOG.md
+            git log ${PREVIOUS_TAG}..${CURRENT_TAG} --pretty=format:"* %s (%h)" >> CHANGELOG.md
+          fi
+
+          echo "" >> CHANGELOG.md
+          echo "" >> CHANGELOG.md
+          echo "## Assets" >> CHANGELOG.md
+          echo "" >> CHANGELOG.md
+          echo "Download the binary for your platform:" >> CHANGELOG.md
+          echo "" >> CHANGELOG.md
+          echo "- **Linux (amd64)**: \`falcon-linux-amd64.tar.gz\`" >> CHANGELOG.md
+          echo "- **Linux (arm64)**: \`falcon-linux-arm64.tar.gz\`" >> CHANGELOG.md
+          echo "- **macOS (Intel)**: \`falcon-darwin-amd64.tar.gz\`" >> CHANGELOG.md
+          echo "- **macOS (Apple Silicon)**: \`falcon-darwin-arm64.tar.gz\`" >> CHANGELOG.md
+          echo "" >> CHANGELOG.md
+          echo "Extract with: \`tar -xzf falcon-{platform}-{arch}.tar.gz\`" >> CHANGELOG.md
+          echo "" >> CHANGELOG.md
+          echo "Verify downloads with \`checksums.txt\` (SHA256):" >> CHANGELOG.md
+          echo "\`\`\`" >> CHANGELOG.md
+          cat release/checksums.txt >> CHANGELOG.md
+          echo "\`\`\`" >> CHANGELOG.md
+
+          cat CHANGELOG.md
```

Comment:
This file isn't actually being committed back to the repository, right? It's just a staging area for the release text?